ladybug_geometry/geometry3d/face.py

```diff
@@ -1151,6 +1151,108 @@ def sub_faces_by_ratio_sub_rectangle(self, ratio, sub_rect_height, sill_height,
+        sill_height_final = (
+            0.01 * rect_height if sill_height < 0.01 * rect_height else sill_height)
+        # adjust sill_height_final if sum of it and sub_face_height_final > rect_height
+        if sub_face_height_final + sill_height_final > rect_height:
+            sill_height_final = rect_height - sub_face_height_final
+
+        max_width_break_up = rect_base / 2
+        num_divisions = (
+            round(rect_base / dist_breakup) if rect_base > dist_breakup / 2 else 1)
+        if (sub_face_width < max_width_break_up and
+            num_divisions * sub_face_width <= rect_base):
+            # divide up the rectangle into points on the bottom
+            div_dist = rect_base / 2 if num_divisions == 1 else dist_breakup
+            if (num_divisions * sub_face_width + (num_divisions - 1) * \
+                (dist_breakup - sub_face_width) > rect_base):
+                num_divisions = math.floor(rect_base / dist_breakup)
+
+            is_odd = lambda num: num % 2
+            start_pt_x = (
+                (num_divisions / 2) * div_dist if is_odd(num_divisions) == 0
+                else (math.floor(num_divisions / 2) + 0.5) * div_dist)
+
+            btm_div_pts = [Point3D(start_pt_x, 0, sill_height_final)]
+            remainder = rect_base - (div_dist * num_divisions)
+            total_dist = remainder / 2
+
+            while total_dist < rect_base:
+                total_dist += div_dist
+                next_pt = (rect_base / 2) - total_dist
+                btm_div_pts.append(Point3D(next_pt, 0, sill_height_final))
+
+            sub_face_lines_start = tuple(LineSegment3D.from_end_points(
+                                         pt, btm_div_pts[i + 1])
+                                         for i, pt in enumerate(btm_div_pts[:-1]))
+            # scale the line segments along their center points
+            line_cent_pt = [
+                Point3D(line.p2.x + ((line.p1.x - line.p2.x) / 2), 0, line.p1.z)
+                for line in sub_face_lines_start
+            ]
+
+            sub_face_line_scale = sub_face_width / div_dist
+
+            sub_face_lines_start = tuple(LineSegment3D.from_end_points(
+                Point3D(a.p1.x - ((a.p1.x - b.x) * (1 - sub_face_line_scale)),
+                        0, a.p1.z),
+                Point3D(a.p2.x + ((b.x - a.p2.x) * (1 - sub_face_line_scale)),
+                        0, a.p1.z))
+                for a, b in zip(sub_face_lines_start, line_cent_pt)
+            )
+            # generate the vertices by 'extruding' along window height vector
+            y_vec = Vector3D(y=sub_face_height_final)
+            final_faces = [Face3D((line.p2, line.p1, line.p1 + y_vec, line.p2 + y_vec),
+                                  base_plane) for line in sub_face_lines_start]
+        else: # make a single sub-rectangle at an appropriate sill height
+            if sub_face_width > rect_base:
+                sub_face_width = rect_base
```

Comment:
I think we need to offset the width a little here since having a sub-face that goes all of the way to the edge of the parent rectangle could break some of the workflows. So this should be:
```
if sub_face_width >= rect_base:
    sub_face_width = rect_base * 0.98
```